pyproject.toml

```diff
@@ -60,12 +60,12 @@ dependencies = [
 [project.optional-dependencies]
 dev = [
```

Comment:
we should consider switching to https://packaging.python.org/en/latest/specifications/dependency-groups/ now that it's supported by both pip and uv

Reply:
(doesn't need to be done in this PR obviously)